src/restartscript.cpp

```diff
@@ -490,7 +493,7 @@ void writeScript(const string& ckptDir,
                   "      else\n"
                   "        DMTCP_REMLAUNCH_0_0=\"$DMTCP_REMLAUNCH_0_0"
                                                      " $DMTCP_LAUNCH_CKPTS\"\n"
-                  "        $srun_path \"$llaunch\"\n"
+                  "        $srun_path --nodes=$srun_ntasks --ntasks=$srun_ntasks \"$llaunch\"\n"
```

Comment:
Yes and if I'm not wrong, it is the expected DMTCP behavior. At restart phase, DMTCP needs one process per node and handles the `fork()`step to recreate per-node process topology. So here, we ensure that the number of needed processes is equal to the number of nodes.

But maybe I misunderstood what DMTCP is doing at restart. Please correct me if I'm wrong